src/NHibernate.Validator/Engine/ClassValidator.cs

```diff
@@ -565,6 +566,175 @@ public IEnumerable<InvalidValue> GetPotentialInvalidValues(string propertyName,
 			return GetPotentialInvalidValues(propertyName, value, null);
 		}
 
+
+		/// <summary>
+		/// Create validators based on hibernate metadata if appropriative validator not defined explicitly
+		/// </summary>
+		/// <param name="properties">Hibernate metadata to analize</param>
+		public void ConfigureFrom(IEnumerable<Property> properties)
+		{
+			foreach(var prop in properties)
+			{
+
+				var propTp = prop.Type.ReturnedClass;
+				var propInfo = entityType.GetProperty(prop.Name, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance); 
+
+				if (!prop.IsNullable 
+					&& (!propTp.IsValueType
+						 || (propInfo != null && propInfo.PropertyType.IsGenericType && propInfo.PropertyType.GetGenericTypeDefinition() == typeof(Nullable<>))))
+				{
+					ConfigurePropValidatotBasedOnMapping<NotNullAttribute>(prop, null);
+				}
+
+				if (propTp== typeof(string))
+				{
+					ConfigurePropValidatotBasedOnMapping<LengthAttribute>(prop
+						, (attr, col) => attr.Max = col.Length);
+				}
+				else if (propTp.IsEnum)
+				{
+					ConfigurePropValidatotBasedOnMapping<EnumAttribute>(prop, null);
+				}
+				else if (   propTp == typeof(decimal) 
+						 || propTp == typeof(double)
+						 || propTp == typeof(float)
+						)
+				{
+					ConfigurePropValidatotBasedOnMapping<DigitsAttribute>(prop
+						, (attr, col) => { 
+											attr.IntegerDigits = col.Precision - col.Scale;
+											attr.FractionalDigits = col.Scale;
+										 } );
+				}
+				else
+				{
+					ConfigureFromChildComponentMapping(prop, propInfo);
+				}

```

Comment:
Formatting would be better like:
```c#
foreach(var prop in properties)
{
	var propTp = prop.Type.ReturnedClass;
	var propInfo = entityType.GetProperty(prop.Name, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance); 

	if (!prop.IsNullable 
		&& (!propTp.IsValueType
			 || (propInfo != null && propInfo.PropertyType.IsGenericType && propInfo.PropertyType.GetGenericTypeDefinition() == typeof(Nullable<>))))
	{
		ConfigurePropValidatotBasedOnMapping<NotNullAttribute>(prop, null);
	}

	if (propTp == typeof(string))
	{
		ConfigurePropValidatotBasedOnMapping<LengthAttribute>(
			prop,
			(attr, col) => attr.Max = col.Length);
	}
	else if (propTp.IsEnum)
	{
		ConfigurePropValidatotBasedOnMapping<EnumAttribute>(prop, null);
	}
	else if (propTp == typeof(decimal)
		|| propTp == typeof(double)
		|| propTp == typeof(float))
	{
		ConfigurePropValidatotBasedOnMapping<DigitsAttribute>(
			prop,
			(attr, col) =>
			{
				attr.IntegerDigits = col.Precision - col.Scale;
				attr.FractionalDigits = col.Scale;
			});
	}
	else
	{
		ConfigureFromChildComponentMapping(prop, propInfo);
	}

	//TODO: Future -> IPropertyConstraint, but how to check this during Update on DB level?

	//TODO: Some extension mechanizm to allow used define rules to auto generate rules
	//      with used defined validators (attributes)
}
```

And following code should similarly be reformatted.

Reply:
Reformatted